File: hubster_app/lib/models/catalog/subscription_service.dart
import 'package:equatable/equatable.dart';

// Represents a predefined subscription service
class SubscriptionService extends Equatable {
  final int id;
  final String name;
  final String? logoUrl;

  const SubscriptionService({
    required this.id,
    required this.name,
    this.logoUrl,
  });

  factory SubscriptionService.fromJson(Map<String, dynamic> json) {
    return SubscriptionService(
      id: json['id'] as int,
      name: json['name'] as String,
      logoUrl: json['logo_url'] as String?,
    );
  }

  Map<String, dynamic> toJson() {
    return {'id': id, 'name': name, 'logo_url': logoUrl};
  }

  @override
  List<Object?> get props => [id, name, logoUrl];
}

File: backend/internal/models/subscription_service.go
package models

import (
	"time"
)

// SubscriptionService defines a type of subscription service offered.
// @name SubscriptionService
type SubscriptionService struct {
	ID        uint      `gorm:"primarykey" json:"id"`
	CreatedAt time.Time `json:"createdAt"`
	UpdatedAt time.Time `json:"updatedAt"`
	Name      string    `gorm:"type:varchar(100);uniqueIndex;not null" json:"name"`
	LogoURL   string    `gorm:"type:text" json:"logo_url,omitempty"`
}

// CreateSubscriptionServiceRequest defines the request body for creating a new subscription service.
// @name CreateSubscriptionServiceRequest
type CreateSubscriptionServiceRequest struct {
	Name    string `json:"name" validate:"required,min=2,max=100"`
	LogoURL string `json:"logo_url" validate:"omitempty,url"`
}
